src/app-layer-dnp3.c

```diff
@@ -1309,20 +1309,10 @@ static int DNP3ParseResponse(Flow *f, void *state, AppLayerParserState *pstate,
     SCReturnInt(-1);
 }
 
-static AppLayerDecoderEvents *DNP3GetEvents(void *state, uint64_t tx_id)
+static AppLayerDecoderEvents *DNP3GetEvents(void *tx)
 {
-    DNP3State *dnp3 = state;
-    DNP3Transaction *tx;
-    uint64_t tx_num = tx_id + 1;
-
-    if (dnp3->curr && dnp3->curr->tx_num == tx_num) {
-        return dnp3->curr->decoder_events;
-    }
-
-    TAILQ_FOREACH(tx, &dnp3->tx_list, next) {
-        if (tx->tx_num == tx_num) {
-            return tx->decoder_events;
-        }
+    if (likely(tx)) {
```

Comment:
in the rust code the tx that is passed is unconditionally dereferenced, so this assumes we never get a null ptr. I think this makes sense. So we can remove checks like this as well.

Reply:
Will do.